src/dataflow/opds1/parse.ts

```diff
@@ -101,6 +107,50 @@ function parseFormat(format: OPDSAcquisitionLink | OPDSIndirectAcquisition) {
   };
 }
 
+/**
+ * Indirect acquisition utils
+ */
+const isIndirectAcquisitionOrType = link =>
+  link?.indirectAcquisitions || link?.type;
+
+const linearizeAcquisitions = ia =>
+  ia.indirectAcquisitions
+    ? ia.indirectAcquisitions.map(acquistion => acquistion.type)
+    : ia.type;
+
+/**
+ * If OPDS entry has more than one acquistion type,
+ * the OPDS acquisition specs specify preserving order of types.
+ * If an entry has "application/epub+zip" followed by "application/pdf",
+ * epub takes precedence. So, get first mimetype.
+ * @see https://github.com/io7m/opds-acquisition-spec?tab=readme-ov-file#linearized-acquisitions
+ */
+function inferEBookFormat(
+  borrowLink: OPDSAcquisitionLink | null
+): BookFormat | undefined {
+  const indirectAcquisitions = borrowLink?.indirectAcquisitions;
+  if (indirectAcquisitions && indirectAcquisitions.length > 0) {
+    const mimeType = indirectAcquisitions
+      .filter(isIndirectAcquisitionOrType)
+      .flatMap(linearizeAcquisitions)
+      .at(0);
+
+    if (mimeType) {
+      if (EPUB_MEDIA_TYPES.includes(mimeType)) {
+        return "ePub";
+      }
+
+      if (mimeType === PdfMediaType) {
+        return "PDF";
+      }
+    }
+
+    return undefined;
+  } else {
+    return undefined;
+  }
```

Comment:
Since we're returning on match I think we can drop the `else` and the `return undefined` inside the `if` and just have a final `return undefined;` at the end of the function.